src/promise-cache.ts

```diff
@@ -48,7 +51,7 @@ export class PromiseCache<T> {
     public get(key: string): Promise<T> {
         const found = this.cache.get(key);
 
-        if (found) {
+        if (found && (this.conf.ttl === "FOREVER" || this.conf.ttlAfter === "ACCESS" || ((found.create + (this.conf.ttl as number)) >= Date.now()))) {
```

Comment:
For me it would be ok to check if the entry should be evicted in both cases ("ACCESS" and "WRITE"). What do you think?

In this case the condition should be something like 
`found && (ttl = "FOREVER" || (ttlAfter = "ACCESS" && checkAccessTime) || (ttAfter = "WRITE" && checkWriteTime))`. 
Right?

I would move the checks after `found &&` into `CacheEntry.isStale(Config config): boolean`. We can then use it in the interval check as well.

If it is stale, we must not forget to actually remove it.